docusaurus/docs/recipes/contentful.md

```diff
@@ -0,0 +1,514 @@
+    }
+
+    render() {
+        return (
+            (...)
+                { this.isPreviewingContentful && <ContentfulPreview> }
+            (...)
+        );
+    }
+}
+```
+
+Using hooks: 
+
+```js
+// App.js
+const App = ({ Component, pageProps, rootSelector, router }) => {
+    const [isPreviewingContentful, setIsPreviewingContentful] = useState(false);
+    
+    // Using a useEffect hook with no dependencies guaranties that it fires only once per instance of App
+    useEffect(() => setIsPreviewingContentful(Object.hasOwnProperty.call(router.query, 'cms-preview')), [])
+
+    return (
+            (...)
+                { this.isPreviewingContentful && <ContentfulPreview> }
+            (...)
+        )
+}
+```
+
+### 4. CMS Translations
+
+This boilerplate already includes **Internationalization** using [`@moxy/next-intl`](https://github.com/moxystudio/next-intl), this is done by statically configuring individual `intl/messages/<locale>.json` per locale. You can do this dynamically by moving this static files to a **Contentful** content model and by enabling localization in it.
+
+#### 1. Defining the content type
+
+Firstly, define the content type that will consist on a list of key/value pairs. Add the fields necessary to fit your needs, for example `homePageTitle`.
+
+> ℹ️ Make sure that you enable localization on every field you set in your content type.
+
+After creating your content type populate the respective fields accordingly.
+
+#### 2. Overriding [`@moxy/next-intl`](https://github.com/moxystudio/next-intl)
+
+Now that you have your content type set you should update the `/intl/index.js` file to fetch the data from **Contentful**. This is done by using the [**Contentful** SDK](https://www.npmjs.com/package/contentful).
+
+```js
+locales: [
+    {
+        id: 'en-US',
+        name: 'English',
+        loadMessages: async () => {
+            const content = await client.getEntries({
+                content_type: 'genericTranslations',
+                locale: 'en-US',
+                // The parameter bellow is needed to avoid the problem described at the top of the file.
+            });
+
+            return content.items[0].fields;
+        },
+    },
+],
+```
+
+## Cache
+
+### 1. Contentful API rate limits
+
+API Rate limits specify the number of requests a client can make to Contentful APIs in a specific time frame. Every request counts against a per second rate limit.
+
+Currently Conentful doesn't enforce any limits on requests that hit their CDN cache. For requests that do hit the Content Delivery API enforces rate limits of 78 requests per second.
+
+When a client gets rate limited, the API responds with the `429 Too Many Requests` status code and sets the `X-Contentful-RateLimit-Reset` header that tells the client when it can make its next request. 
+
+### 2. Custom Caching Layer
+
+One preventive measure for avoiding hitting the rate limit for Contentful is to implement your own custom caching layer.
```

Comment:
> One preventive measure for avoiding hitting the rate

This sentence doesn't sound good. I would change to "One preventive measure to avoid hitting the rate".